pkg/orchestration/wiring/kubeingress/plugin_test.go

```diff
@@ -5,14 +5,94 @@ import (
 
 	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
 	"github.com/atlassian/voyager"
+	"github.com/atlassian/voyager/pkg/apis/orchestration/v1"
 	"github.com/atlassian/voyager/pkg/k8s"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/k8scompute/api"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
 	"github.com/stretchr/testify/assert"
 	apps_v1 "k8s.io/api/apps/v1"
+	ext_v1b1 "k8s.io/api/extensions/v1beta1"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+func TestBuildingIngressResource(t *testing.T) {
+	t.Parallel()
+
+	var serviceName smith_v1.ResourceName = "serviceName"
+
+	validOutput := wiringplugin.WiredSmithResource{
```

Comment:
You are using subtest here (which is fine by itself, of course) but I'd rather use just normal tests (lift them up) and a helper function to generate this resource/etc. In general I think it is better to not share state between tests to avoid and potential interference (including caused by bugs in tests) between them.

Reply:
I've pulled the fixture setup out into a function.